CreateServer.ps1

```diff
@@ -240,62 +222,141 @@ Function Logout {
+    }
+
+    if ($Simple) {
+        $script:Context.NoProxy = $true
+        $script:Context.NoPki = $true
+    }
+}
+
 Function Initialize-Variables {
-    $script:VmName = $VmName.ToLower()
-    $script:Account = (az account show | ConvertFrom-Json)
-    $script:Subscription = $Account.id
+    $script:Context.VmName = $VmName.ToLower()
+    $script:Context.ProxyVmName = $Context.VmName + "-server"
+    $script:Context.Location = $Location
+    $script:Context.Environment = $Environment
+    $script:Context.Username = $Username
+    $script:Context.Size = $Size
+    $script:Context.ProxySize = $ProxySize
+    $script:Context.Platform = $Platform
+    $script:Context.BundleIds = $BundleIds
+    $script:Context.Username = $Username
+    $script:Context.AuthenticatedProxyCredentials = $AuthenticatedProxyCredentials
+    $script:Context.IncludeRoutes = $IncludeRoutes
+    $script:Context.ExcludeRoutes = $ExcludeRoutes
+    $script:Context.ListenPort = $ListenPort
+    $script:Context.NoProxy = $NoProxy
+    $script:Context.NoPACUrl = $NoPACUrl
+    $script:Context.UseInspection = $UseInspection
+    $script:Context.UseAllowList = $UseAllowList
+
+    $script:Context.Account = (az account show | ConvertFrom-Json)
+    $script:Context.Subscription = $Context.Account.id
     if ($Email -eq "") {
         Write-Header "Email not provided. Detecting email..."
-        $script:Email = $Account.user.name
-        Write-Host "Detected your email as '$Email'"
+        $script:Context.Email = $Context.Account.user.name
+        Write-Host "Detected your email as '$($Context.Email)'"
+    }
+    else {
+        $script:Context.Email = $Email
     }
 
-    $script:ResourceGroup = "$VmName-group"
-    $script:SSHKeyPath = "$HOME/.ssh/$VmName"
+    $script:Context.ResourceGroup = "$VmName-group"
+    $script:Context.SSHKeyPath = "$HOME/.ssh/$VmName"
 
-    $script:GraphContext = Get-MgContext
+    $script:Context.GraphContext = Get-MgContext
 
-    $script:FQDN = "$VmName.$Location.cloudapp.azure.com"
+    $script:Context.TunnelFQDN = "$VmName.$Location.cloudapp.azure.com"
+    $script:Context.ServiceFQDN = "$VmName-server.$Location.cloudapp.azure.com"
 
     if ($PACUrl -eq "") {
-        $script:PACUrl = "http://$FQDN/tunnel.pac"
+        $script:Context.PACUrl = "http://$($Context.TunnelFQDN)/tunnel.pac"
+    }
+    else {
+        $script:Context.PACUrl = $PACUrl
     }
 
     if (-Not $Delete -And -Not $SprintSignoff) {
         # We only need a group name for the create and profile flows
-        $script:Group = Get-MgGroup -Filter "displayName eq '$GroupName'"
-        if (-Not $Group) {
+        $script:Context.Group = Get-MgGroup -Filter "displayName eq '$GroupName'"
+        if (-Not $Context.Group) {
             Write-Error "Could not find group named '$GroupName'"
             Exit -1
         }
+        $script:Context.GroupName = $GroupName
     }
 }
 
 Function New-Profiles {
-    if ($NoProxy) {
+    if ($Context.NoProxy) {
```

Comment:
Since we now have a central context, we could probably move things like ProxyHostname, ProxyPort and PACUrl to be computed variables.